File: 1 Python/solutions/blackjack.py
def handle_card(ordinal: str, hands: list) -> list:
    card = input(f'What is your {ordinal} card? ').lower()
    if card in ['j', 'q', 'k']:
        hands = list(map(lambda c: c + 10, hands))
    elif card == 'a':
        for i in range(len(hands)):
            val = hands[i]
            hands[i] += 1
            hands.append(val+11)
    else:
        try:
            if int(card) >= 2 and int(card) <= 10:
                hands = list(map(lambda c: c + int(card), hands))
            else:
                print('That is not a valid card, please try again')
                handle_card(ordinal, hands)
        except ValueError:
            print('That is not a valid card, please try again')
            handle_card(ordinal, hands)
    return hands


def advise(hands: list) -> tuple:
    if 21 in hands:
        return 'BLACKJACK!', 21
    elif min(hands) > 21:
        return 'BUSTED', min(hands)

    max_val = max(filter(lambda c: c < 21, hands))
    if max_val >= 17:
        return 'stay', max_val
    else:
        return 'hit', max_val


if __name__ == '__main__':
    first = handle_card('first', [0])
    second = handle_card('second', first)
    third = handle_card('third', second)
    advice = advise(third)
    print(f'With {advice[1]} points, {advice[0]}')

File: code/liam/js/lab01-03/partA/blackjack.js
alert('Blackjack suggestions :)')
let firstCard = prompt('What is your first card?').toUpperCase()
let secondCard = prompt('What is your second card?').toUpperCase()
let thirdCard = prompt('What is your third card?').toUpperCase()

const cardWorth = {
    'A': 1,
    'J': 10,
    'Q': 10,
    'K': 10,
    '2': 2,
    '3': 3,
    '4': 4,
    '5': 5,
    '6': 6,
    '7': 7,
    '8': 8,
    '9': 9,
}

let firstValue = cardWorth[firstCard]
let secondValue = cardWorth[secondCard]
let thirdValue = cardWorth[thirdCard]

console.log(firstValue, secondValue, thirdValue)

let handValues = [firstValue, secondValue, thirdValue]
let hand = [firstCard, secondCard, thirdCard]
let total = handValues.reduce((adding, a) => adding + a, 0)

alert(`Got it- You have ${firstCard}, ${secondCard}, and ${thirdCard}`)

console.log(total)

if (hand.includes('A')) {
    if (total <= 11) {
        total += 10
    }
}

if (firstValue == undefined || secondValue == undefined || thirdValue == undefined) {
    alert(`Invalid entry included. Please try again.`)
} else if (total < 17) {
    alert(`At ${total}, you should hit`)
} else if (total >= 17 && total < 21) {
    alert(`With ${total}- stay.`)
} else if (total === 21) {
    alert(`${total}!!! Blackjack!`)
} else {
    alert(`At ${total}, you've already busted.`)
}